docs/docs/reference/unused-terms.md

```diff
@@ -0,0 +1,146 @@
+```
+
+What happens with unused values at runtime?
+-------------------------------------------
+As `unused` are guaranteed not to be used in computations, they can and will be erased.
+
+```scala
+// becomes def methodWithUnusedEv(): Int at runtime
+def methodWithUnusedEv(unused ev: Ev): Int = ...  
+
+def evidence1: Ev = ...
+unused def unusedEvidence2: Ev = ... // does not exist at runtime
+unused val unusedEvidence3: Ev = ... // does not exist at runtime
+
+// evidence1 is not evaluated and no value is passed to methodWithUnusedEv
+methodWithUnusedEv(evidence1)
+```
+
+State machine with unused evidence example
+------------------------------------------
+The following examples is an extended implementation of a simple state machine which can be in a state `On` or `Off`.
+The machine can change state from `Off` to `On` with `turnedOn` only if it is currently `Off`, 
+conversely from `On` to `Off` with `turnedOff` only if it is currently `On`. These last constraint are
+captured with the `IsOff[S]` and `IsOn[S]` implicit evidence only exist for `IsOff[Off]` and `InOn[On]`. 
+For example, not allowing calling `turnedOff` on in an `Off` state as we would require an evidence `IsOn[Off]` 
+that will not be found.
+
+As the implicit evidences of `turnedOn` and `turnedOff` are not used in the bodies of those functions 
+we can mark them as `unused`. This will remove the evidence parameters at runtime, but we would still 
+evaluate the `isOn` and `isOff` implicits that where found as arguments.
+As `isOn` and `isOff` are not used except as as `unused` arguments, we can mark them as `unused`, hence 
+removing the evaluation of the `isOn` and `isOff` evidences.
+
+```scala
+import scala.annotation.implicitNotFound
+
+sealed trait State
+final class On extends State
+final class Off extends State
+
+@implicitNotFound("State is must be Off")
+class IsOff[S <: State]
+object IsOff {
+  // def isOff will not be called at runtime for turnedOn, the compiler will only require that this evidence exists
+  implicit def isOff: IsOff[Off] = new IsOff[Off]
+}
+
+@implicitNotFound("State is must be On")
+class IsOn[S <: State]
+object IsOn {
+  // def isOn will not exist at runtime, the compiler will only require that this evidence exists at compile time
+  unused implicit val isOn: IsOn[On] = new IsOn[On]
+}
+
+class Machine[S <: State] private {
+  // ev will disapear from both functions
+  def turnedOn(implicit unused ev: IsOff[S]): Machine[On] = new Machine[On]
+  def turnedOff(implicit unused ev: IsOn[S]): Machine[Off] = new Machine[Off]
+}
+
+object Machine {
+  def newMachine(): Machine[Off] = new Machine[Off]
+}
+
+object Test {
+  def main(args: Array[String]): Unit = {
+    val m = Machine.newMachine()
+    m.turnedOn
+    m.turnedOn.turnedOff
+
+    // m.turnedOff
+    //            ^
+    //            State is must be On
+
+    // m.turnedOn.turnedOn
+    //                    ^
+    //                    State is must be Off
+  }
```

Comment:
Need to also give a summary of the rules that govern unused parameters at the end of the section. If something is too complicated (I hope not!) need to refer to a separate issue that explains the rules in detail.